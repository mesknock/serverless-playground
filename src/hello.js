// TODO: remove as just generic example

import fetch from 'node-fetch';
import { successResponse, runWarm } from './utils';
import foo from './services/foo';

const hello = async (/* event, context */) => {
  // // successResponse handles wrapping the response in an API Gateway friendly
  // // format (see other responses, including CORS, in `./utils/lambda-response.js)
  // const response = successResponse({
  //   message: 'Yeah, baby! Go Serverless! Your function executed successfully!',
  //   input: event,
  // });
  // callback(null, response);

  const breweries = await fetch('https://api.openbrewerydb.org/breweries').then(res => res.json())
  const names = foo(breweries);
  return successResponse({ names });
};

// runWarm function handles pings from the scheduler so you don't
// have to put that boilerplate in your function.
export default runWarm(hello);
