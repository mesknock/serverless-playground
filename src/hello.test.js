import hello from './hello';

describe('hello', () => {
  it('executes as expected', async () => {
    const cb = jest.fn();
    await hello({}, {}, cb);
    expect(cb).toBeCalled();
    expect(cb).toMatchSnapshot();
  });
});
