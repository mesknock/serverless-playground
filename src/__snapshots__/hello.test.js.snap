// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello executes as expected 1`] = `
Object {
  "body": "{\\"names\\":[\\"Avondale Brewing Co\\",\\"Band of Brothers Brewing Company\\",\\"Trim Tab Brewing\\",\\"Yellowhammer Brewery\\",\\"Bearpaw River Brewing Co\\",\\"King Street Brewing Co\\",\\"1912 Brewing\\",\\"Bad Water Brewing\\",\\"BJs Restaurant & Brewery - Chandler\\",\\"BlackRock Brewers\\",\\"Dragoon Brewing Co\\",\\"Grand Canyon Brewing Company\\",\\"Mudshark Brewing Co\\",\\"Richter Aleworks\\",\\"SanTan Brewing Co\\",\\"State 48 Brewery\\",\\"Wren House Brewing Company\\",\\"Brick Oven Pizza Co / Brick & Forge Brewing\\",\\"Diamond Bear Brewing Co\\",\\"Lost Forty Brewing\\"]}",
  "statusCode": 200,
}
`;
